### PR TITLE
Add pieces

### PR DESCRIPTION
**Pull request by barlock**
**Related Issue**  
Supports #4 

**Related PRs**  
This PR is not dependent on any other PR

**What does this PR do?**  
When a player joins a game, it adds their 20 pieces to the board. 

You can now open 4 browser windows and add 4 players and watch them come in on all of them! 

It's _realllllyyy_ slow. I don't think ganache can handle that many websocket subscriptions 😅. I'll try to switch to Pantheon as we should maybe be doing that anyway and see if that makes a difference. 

**Description of Changes**  
Lots of work in the sagas and initializing all the things. 

**What gif most accurately describes how I feel towards this PR?**

### CODE REVIEW

File: packages/stratego-four/contracts/Stratego4.sol [hunk 1]
```diff
@@ -12,8 +12,8 @@ contract Stratego4 {
 
   struct Player {
     address player;
-    uint256[] pieceRankHashes;
-    mapping(uint256 => Piece) pieces;
+    bytes32[] pieceRankHashes;
```
Comment (barlock) [inline packages/stratego-four/contracts/Stratego4.sol]: sha3 is bytes32 not uint256 🤷‍♂ I learned something.

File: packages/stratego-four/contracts/Stratego4.sol [hunk 2]
```diff
@@ -26,8 +26,8 @@ contract Stratego4 {
   mapping (uint => Game) public games;
   mapping (address => uint) public playerGames;
 
-  function currentGame() public view returns(uint) {
-    return playerGames[msg.sender];
+  function currentGame(address player) public view returns(uint) {
```
Comment (barlock) [inline packages/stratego-four/contracts/Stratego4.sol]: Passed in a `player` for drizzle's caching

File: packages/stratego-four/package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "start": "concurrently -n w: npm:start:*",
     "start:ganache": "yarn ganache-cli -i 42 -d",
     "start:web": "react-scripts start",
-    "start:migrate": "truffle migrate --reset --skip-dry-run --network=development --contracts_build_directory=$(pwd)/src/build/contracts",
+    "start:migrate": "rm -rf src/build && truffle migrate --reset --skip-dry-run --network=development --contracts_build_directory=$(pwd)/src/build/contracts",
```
Comment (barlock) [inline packages/stratego-four/package.json]: requires you to refresh the page on first load... but ganache was having issues with old build files preventing a recompile

File: packages/stratego-four/src/game/gameReducer.js [hunk 4]
```diff
@@ -1,101 +1,11 @@
...
-    ]
-  }
+  cache: null,
+  player: null
 };
 
 export const JOIN_GAME = 'stratego/game/JOIN_GAME';
 export const SET_PLAYER = 'stratego/game/SET_PLAYER';
-export const SET_CURRENT_GAME_KEY = 'stratego/game/SET_CURRENT_GAME_KEY';
+export const SET_CACHE_KEY = 'stratego/game/SET_CACHE_KEY';
```
Comment (barlock) [inline packages/stratego-four/src/game/gameReducer.js]: Made this generic as I use the pattern a lot

File: packages/stratego-four/src/game/gameSaga.js [hunk 5]
```diff
@@ -24,13 +31,23 @@ function* defaultTxOps() {
...
  */
-function* initializePlayer(player) {
-  yield put(setPlayer(player));
-
+function* initializePlayer({ player }) {
   // get the ops after setting the player
   const ops = yield defaultTxOps();
 
-  yield put(setCurrentGameKey(stratego4.methods.currentGame.cacheCall(ops)));
+  yield put(
```
Comment (breakpointer) [inline packages/stratego-four/src/game/gameSaga.js]: TIL; javascript has a `yield` 🤔 [I read the docs... ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield) neat! 👍

File: packages/stratego-four/src/game/gameSaga.js [hunk 6]
```diff
@@ -49,7 +66,127 @@ export function initializeWeb3(action) {
...
   const accounts = yield select(selectAccounts);
-  yield initializePlayer(accounts[0]);
+
+  yield put(setPlayer(accounts[0]));
+}
+
+/**
+ * Initialize player pieces with default values
+ */
+export function* initializePlayerPieces() {
```
Comment (breakpointer) [inline packages/stratego-four/src/game/gameSaga.js]: ...and I was wondering how the function caller would know they were actually getting a `Generator` type object/function....
Another TIL! [Good docs too 👍](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*)

Comment (breakpointer) [reply packages/stratego-four/src/game/gameSaga.js]: mind blown

File: packages/stratego-four/src/game/gameSaga.js (hunk 5 above)
Comment (barlock) [reply packages/stratego-four/src/game/gameSaga.js]: Yeah it was another attempt at async before async/await. I expect they'll rewrite this library to use that syntax soonish?

Also, interesting fact. `async/wait` is polified on browsers with generators. hence the name "regenerator runtime"

**Pull request merged**